src/main/java/eu/dissco/core/translator/component/SourceSystemComponent.java

```diff
@@ -22,5 +23,8 @@ public SourceSystemComponent(ApplicationProperties applicationProperties,
     }
     this.sourceSystemName = sourceSystemInformation.sourceSystemName();
     this.sourceSystemEndpoint = sourceSystemInformation.sourceSystemUrl();
+    var stringBuilder = new StringBuilder();
```

Comment:
Maybe add a check here if (!filter.empty) {}
Then we don't need to init the builder if not necessary